Kernel/Locking/ProtectedValue.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/Locking/ContendedResource.h>
+
+namespace Kernel {
```

Comment:
Should this go into `AK` so it's potentially usable from usermode and kernel? I don't see anything kernel specific about it?

Reply:
It ultimately depends on `Kernel::Mutex` for the actual locking primitive. If that could be either imported into AK or templated away, then there's nothing preventing this from being moved out of the kernel.